src/main.c

```diff
@@ -534,11 +534,29 @@ void dict(struct state *s)
 			}
 		} else if (strcmp(subfunc, "auth") == 0) {
 			if (peek_arg(&(s->cmd_list)) == NULL) {
-				printf("No username specified.\n");
-			} else {
+				char filename[] = "user.inf";
+				FILE*f;
+				int len;
+				rsp = exword_get_file(s->device, basename(filename), &buffer, &len);
+				if (rsp == 0x20)
+					rsp = write_file(filename, buffer, len);
+				free(buffer);
+				if ((f = fopen("user.inf", "r")) == NULL){
+					user = NULL;
+					printf("No username specified.\n");
+				}
+				else{
+					user = xmalloc(32);
+					fscanf(f, "%s\n", user);
+					fclose(f);
+				}
+			}
+			else {
 				user = xmalloc(strlen(peek_arg(&(s->cmd_list)) + 1));
 				strcpy(user, peek_arg(&(s->cmd_list)));
 				dequeue_arg(&(s->cmd_list));
+			}
+			if(user != NULL){
```

Comment:
Please add a space between the if and opening paren and the closing paren and opening curly brace

> if (user != NULL) {